dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/JNDIOutputTest.java

```diff
@@ -0,0 +1,231 @@
+        assertNotNull("Security service did not report it was ready",
+                      libertyServer.waitForStringInLog("CWWKS0008I"));
+        assertNotNull("Server did not came up",
+                      libertyServer.waitForStringInLog("CWWKF0011I"));
+
+        Log.info(c, "setUp", "Creating servlet connection the server");
+        servlet = new UserRegistryServletConnection(libertyServer.getHostname(), libertyServer.getHttpDefaultPort());
+
+        if (servlet.getRealm() == null) {
+            Thread.sleep(5000);
+            servlet.getRealm();
+        }
+
+        /*
+         * The original server configuration has no registry or Federated Repository configuration.
+         */
+        emptyConfiguration = libertyServer.getServerConfiguration();
+    }
+
+    /**
+     * Configure the embedded LDAP server.
+     *
+     * @throws Exception If the server failed to start for some reason.
+     */
+    private static void setupldapServer() throws Exception {
+        ds = new InMemoryLDAPServer(SUB_DN);
+
+        Entry entry = new Entry(SUB_DN);
+        entry.addAttribute("objectclass", "top");
+        entry.addAttribute("objectclass", "domain");
+        ds.add(entry);
+        /*
+         * Add the partition entries.
+         */
+        entry = new Entry("ou=Test,o=ibm,c=us");
+        entry.addAttribute("objectclass", "organizationalunit");
+        entry.addAttribute("ou", "Test");
+        ds.add(entry);
+
+        entry = new Entry(USER_BASE_DN);
+        entry.addAttribute("objectclass", "organizationalunit");
+        entry.addAttribute("ou", "Test");
+        entry.addAttribute("ou", "TestUsers");
+        ds.add(entry);
+
+        entry = new Entry(USER_DN);
+        entry.addAttribute("objectclass", "inetorgperson");
+        entry.addAttribute("uid", USER);
+        entry.addAttribute("sn", USER);
+        entry.addAttribute("cn", USER);
+        entry.addAttribute("userPassword", PWD);
+        ds.add(entry);
+
+    }
+
+    /**
+     * Convenience method to configure the Liberty server with an {@link LdapRegistry} configuration that
+     * will connect to {@link #ldapServer}.
+     *
+     * @throws Exception If there was an error configuring the server.
+     */
+    private static void updateLibertyServer() throws Exception {
+        ServerConfiguration server = emptyConfiguration.clone();
+
+        LdapRegistry ldap = new LdapRegistry();
+        ldap.setId("ldap1");
+        ldap.setRealm("LDAPRealm");
+        ldap.setHost("localhost");
+        ldap.setPort(String.valueOf(ds.getLdapPort()));
+        ldap.setBaseDN(SUB_DN);
+        ldap.setBindDN(InMemoryLDAPServer.getBindDN());
+        ldap.setBindPassword(InMemoryLDAPServer.getBindPassword());
+        ldap.setLdapType("Custom");
+        ContextPool cp = new ContextPool(true, 0, 2, 1, "2s", "5s");
+        ldap.setContextPool(cp);
+        SearchResultsCache src = new SearchResultsCache();
```

Comment:
probably should disable attributes cache as well. You want to hit your BEROutputStream as much as you can in the test.